python/ray/tune/test/trial_runner_test.py

```diff
@@ -559,6 +559,27 @@ def _restore(self, state):
             self.assertEqual(trial.status, Trial.TERMINATED)
             self.assertTrue(trial.has_checkpoint())
 
+    def testIterationCounter(self):
+        def train(config, reporter):
+            for i in range(config["iterations"]):
+                reporter(itr=i, done=i == config["iterations"] - 1)
+
+        register_trainable("exp", train)
+        config = {
+            "my_exp": {
+                "run": "exp",
+                "config": {
+                    "iterations": 100,
+                },
+                "stop": {
+                    "timesteps_total": 100
+                },
+            }
+        }
+        [trial] = run_experiments(config)
+        self.assertEqual(trial.status, Trial.TERMINATED)
+        self.assertEqual(trial.last_result[TRAINING_ITERATION], 100)
```

Comment:
wouldn't this be 99? perhaps again try running this locally with `pytest trial_runner_test.py -k IterationCounter`